plugins/module_utils/dnac.py

```diff
@@ -553,6 +554,31 @@ def check_status_api_events(self, status_execution_id):
 
         return events_response
 
+    def is_valid_server_address(self, server_address):
+        """
+        Validates the server address to check if it's a valid IPv4, IPv6 address, or a valid hostname.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            server_address (str): The server address to validate.
+        Returns:
+            bool: True if the server address is valid, otherwise False.
+        """
+        # Check if the address is a valid IPv4 or IPv6 address
+        try:
+            ipaddress.ip_address(server_address)
+            return True
+        except ValueError:
+            pass
+
+        # Define the regex for a valid hostname
+        hostname_regex = re.compile(r'^(?!-)[A-Za-z0-9-]{1,63}(?<!-)(\.[A-Za-z0-9-]{1,63})*(\.[A-Za-z]{2,6})?$')
```

Comment:
Can you add with more comments?

```
    hostname_regex = re.compile(
        r'^(?!-)'  # Hostname must not start with a hyphen
        r'[A-Za-z0-9-]{1,63}'  # Hostname segment must be 1-63 characters long
        r'(?!-)$'  # Hostname segment must not end with a hyphen
        r'(\.[A-Za-z0-9-]{1,63})*'  # Each segment can be 1-63 characters long
        r'(\.[A-Za-z]{2,6})$'  # Top-level domain must be 2-6 alphabetic characters
    )
```

Reply:
Added inline examples with comments.

Reply:
By mistake I added more comments in the regex_pattern defined in Webhook destination. Update the comments for hostname_regex in dnac.py.